File: app/src/main/java/zone/com/zadapter3/adapter/RightDelegates.java
package zone.com.zadapter3.adapter;

import com.zone.adapter3.bean.Holder;
import com.zone.adapter3.bean.ViewDelegates;

import zone.com.zadapter3.R;

/**
 * [2017] by Zone
 */

public class RightDelegates extends ViewDelegates<String> {


    @Override
    public int getLayoutId() {
        return R.layout.item_right;
    }

    @Override
    public void fillData(int postion, String data, Holder holder) {
        holder.setText(R.id.tv, data);
    }

    @Override
    public boolean isFullspan() {
        return true;
    }
}

File: app/src/main/java/zone/com/zadapter3kt/adapter/RightDelegates.kt
package zone.com.zadapter3kt.adapter

import com.zone.adapter3kt.data.DataWarp

import zone.com.zadapter3.R
import zone.com.zadapter3kt.adapterimpl.HolderExDemoImpl
import zone.com.zadapter3kt.adapterimpl.ViewDelegatesDemo

/**
 * [2017] by Zone
 */

class RightDelegates : ViewDelegatesDemo<String>() {
    override val layoutId: Int = R.layout.item_right


    override fun onBindViewHolder(position: Int, item: DataWarp<String>, baseHolder: HolderExDemoImpl, payloads: List<*>) {
        baseHolder.setText(R.id.tv, item.data!!)
    }
}
